src/workerd/api/tests/compat-flags-test.js

```diff
@@ -0,0 +1,50 @@
+import {
+  ok,
+  strictEqual,
+  throws,
+} from 'node:assert';
+
+const kOn = 'on';
+const kOff = 'off';
```

Comment:
why not boolean? this makes the API incompatible with current tree-shaking tools. could we use booleans please?

Reply:
Can you explain why I boolean would make a difference for tree-shaking?

Are you saying this would be tree-shaken:

```js
if (globalThis.Cloudflare?.compatibilityFlags.nodejs_compat_v2)
```

But this wouldn't?

```js
if (globalThis.Cloudflare?.compatibilityFlags.nodejs_compat_v2 === "on")
```

Reply:
> ... why not boolean?

Largely due to @kentonv's argument about being able to tell the difference between a disabled flag vs. a non-existent one. Specifically, using a boolean encourages code like `if (!compatFlags['foo']) { ... }` which will be indistinguishable between `undefined` and explicitly `false`. By ensuring that flags that are recognized-but-explicitly-off still use a truthy value we make these checks more reliable and explicit. Only non-existent flags will be `undefined`.

The only other way to address this would be to explicitly throw an error for non-existent flags, which requires more boilerplate and complexity to get right (module developers would need to worry about an additional error being thrown)

Reply:
I should clarify that string values are supported as well by rollup and esbuild: https://esbuild.github.io/api/#define

However they are more prone to coding patterns that are not (reliably) tree-shakable for example:

```
if (globalThis.Cloudflare?.compatibilityFlags.nodejs_compat_v2 === "o" + "n")
```

or 

```
if (globalThis.Cloudflare?.compatibilityFlags.nodejs_compat_v2 === onFlag())
```

Additionally, the string values literally translate to `true` and `false`, and the flag system is designed  with `boolean`s at heart (hence the `no_` support). So using string `'on'` and `'off'` seems just to add no benefits, and only makes for a more verbose and clunky API.

Reply:
https://esbuild.github.io/try/#dAAwLjIzLjAALS1kZWZpbmU6Z2xvYmFsVGhpcy5DbG91ZGZsYXJlLmNvbXBhdGliaWxpdHlGbGFncy5mb289IidvbiciIC0tdHJlZS1zaGFraW5nPXRydWUAZnVuY3Rpb24gZm9vKCkge30KZnVuY3Rpb24gYmFyKCkge30KCmlmIChnbG9iYWxUaGlzLkNsb3VkZmxhcmUuY29tcGF0aWJpbGl0eUZsYWdzLmZvbyA9PT0gJ29uJykgewogIGZvbygpOwp9IGVsc2UgewogIGJhcigpOwp9 illustrates that tree shaking is certainly possible with esbuild with the string values. Yes, there are ways to write the code that causing tree shaking to fail that's also possible with a `boolean` value also.